.github/workflows/win64_llvm_clang_build_and_cache.yml

```diff
@@ -0,0 +1,54 @@
+name: Build CMake Libraries
+
+on:
+  workflow_dispatch:
+
+env:
+    LLVM_VER: 16.0.4
+    BRANCH: release/16.x
+
+jobs:
+
+   build:
+    runs-on: windows-latest
+ 
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup Ninja
+      uses: ashutoshvarma/setup-ninja@master
+      with:
+        # ninja version to download. Default: 1.10.0
+        version: 1.10.0
+        dest: ${{ github.workspace }}/ninja_bin
+               
+    - name: Clone LLVM
+      uses: actions/checkout@v3
+      with:
+        repository: 'llvm/llvm-project'
+        ref: ${{ env.BRANCH }}
+        path: 'llvm-project-${{ env.LLVM_VER }}'
+        fetch-depth: 1
+
+
+    - name: Configure CMake
+      shell: cmd
+      run: |
+        "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64 &^
+        cd llvm-project-%LLVM_VER% &^
+        mkdir build &^
+        cd build &^
+        cmake -GNinja -DCMAKE_CXX_COMPILER=clang-cl -DCMAKE_C_COMPILER=clang-cl -DLLVM_ENABLE_PROJECTS=clang -DCMAKE_BUILD_TYPE=Release -DCMAKE_EXE_LINKER_FLAGS="/machine:x64" -DCMAKE_INSTALL_PREFIX="%GITHUB_WORKSPACE%/llvm-project-%LLVM_VER%/install" ..\llvm
+
+    - name: Build
+      shell: cmd
+      run: |
+        "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" x64 &^
+        cd llvm-project-%LLVM_VER%/build &^
+        ninja clang install
+
+    - name: Cache LibClang
+      id: cache-clang
+      uses: actions/cache@v3
+      with:
+        path: 'llvm-project-${{ env.LLVM_VER }}/install'
+        key: 'libclang-${{ runner.os }}-${{ github.run_id }}'

```

Comment:
Don't cache based on `github.run-id`, cache based on the llvm version. That way we can pull it down (restore) from other workflows in a reasonably memorable way. The run-id is apparently just some random number assigned to the workflow that's unique for the repository

>  A unique number for each workflow run within a repository. This number does not change if you re-run the workflow run